### PR TITLE
Key generation prep

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: src/policy.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            TouchPolicy::Default => 0,
+            TouchPolicy::Never => 1,
+            TouchPolicy::Always => 2,
+            TouchPolicy::Cached => 3,
+        }
+    }
+}
+
+impl TouchPolicy {
+    /// Writes the `TouchPolicy` in the format the YubiKey expects during key generation
```
Comment (str4d) [inline src/policy.rs]: I realised after making this branc that "the format the YubiKey expects" is simply TLV. Maybe there's a nicer way to abstract it, but for now this is probably fine.

File: src/key.rs [hunk 2]
```diff
@@ -460,32 +469,21 @@ pub fn generate(
 
     let txn = yubikey.begin_transaction()?;
 
-    templ[3] = slot.into();
+    let templ = [0, Ins::GenerateAsymmetric.code(), 0, slot.into()];
```
Comment (tarcieri) [inline src/key.rs]: Sidebar but: wonder if we could come up with a better representation for these templates

File: src/mgm.rs [hunk 3]
```diff
@@ -268,7 +268,7 @@ impl MgmKey {
     pub(crate) fn decrypt(&self, input: &[u8; DES_LEN_DES]) -> [u8; DES_LEN_DES] {
         let mut output = input.to_owned();
         TdesEde3::new(GenericArray::from_slice(&self.0))
-            .encrypt_block(GenericArray::from_mut_slice(&mut output));
+            .decrypt_block(GenericArray::from_mut_slice(&mut output));
```
Comment (tarcieri) [inline src/mgm.rs]: Oops, that's likely my bad

**Comment by tarcieri** [comment]
Looks great, thanks!

**Pull request merged**